### ISSUE TITLE
Python SSHNPD: Forward Tunnel SSH Connection

### ISSUE THREAD

**Issue by Xlin123** [atsign-foundation/noports#400]
@gkc I'll make a meeting to discuss further.

**Comment by gkc** [comment]
When using noports setting up the tunnel with reverse ssh (reverse ssh being the only approach supported before v4), the behaviour can be summarized as:
- Client cuts ephemeral ssh keys, and stores the ephemeral public key in its authorized_hosts file
- Client contacts rvd to ask for host and two ports
- Client starts an sshrv session which bridges $rvdHost:$rvdPort1 to the **client's** localhost:22
- Client sends notification to sshnpd to (1) share ephemeral ssh private key with the sshnpd and (2) share details of the rvd host and port to use
- Using those ephemeral keys the sshnpd can ssh to the client host (via $rvdHost:$rvdPort2 which is bridged to the client's port 22)
  - and requests a **remote** port forwarding from $someLocalPort on the client host to the device host's port 22
  `-R 12345:localhost:22`
- sshnpd sends 'connected' notification to the client
- Using that ssh tunnel, users may now `ssh -p $someLocalPort $user@localhost` and the ssh connection is actually being made to port 22 on the device

When using noports setting up the tunnel with forward ssh (v4+) the behaviour can be summarized as:
- Client contacts rvd to ask for host and two ports
- Client sends notification to sshnpd to share details of the rvd host and port to use
- sshnpd starts an sshrv session which bridges $rvdHost:$rvdPort2 to the **device's** localhost:22
- sshnpd sends 'connected' notification to client
- Client can now ssh to the device host (via $rvdHost:$rvdPort1 which is bridged to the device's port 22)
  - and requests a **local** port forwarding from $someLocalPort on the client host to the device host's port 22
    `-L 12345:localhost:22`
- Using that ssh tunnel, users may now `ssh -p $someLocalPort $user@localhost` and the ssh connection is actually being made to port 22 on the device

To summarize even further:
- Tunnel via reverse ssh
  - client runs sshrv
  - tunnel is set up by daemon ssh'ing to client via rvd, and requesting 'remote' port forwarding
- Tunnel via forward ssh
  - daemon runs sshrv
  - tunnel is set up by client ssh'ing to daemon device via rvd, and requesting 'local' port forwarding

The "forward ssh" is described in detail here https://github.com/atsign-foundation/sshnoports/blob/trunk/docs/UnderTheHood.md
